apps/wallet/src/ui/app/shared/faucet/FaucetRequestButton.tsx

```diff
@@ -15,32 +17,40 @@ export type FaucetRequestButtonProps = {
     trackEventSource: 'home' | 'settings';
 };
 
+const debounceWaitInMilliseconds = 2000;
+
 function FaucetRequestButton({
     variant = 'primary',
     trackEventSource,
 }: FaucetRequestButtonProps) {
     const network = useAppSelector(({ app }) => app.apiEnv);
     const networkName = API_ENV_TO_INFO[network].name.replace(/sui\s*/gi, '');
-    const mutation = useFaucetMutation();
+    const { isMutating, mutateAsync, enabled } = useFaucetMutation();
+
+    const onClick = useCallback(() => {
+        toast.promise(mutateAsync(), {
+            loading: <FaucetMessageInfo loading />,
+            success: (totalReceived) => (
+                <FaucetMessageInfo totalReceived={totalReceived} />
+            ),
+            error: (error) => <FaucetMessageInfo error={error.message} />,
+        });
+
+        trackEvent('RequestGas', {
+            props: { source: trackEventSource, networkName },
+        });
+    }, [mutateAsync, networkName, trackEventSource]);
+
+    const debouncedOnClick = useMemo(
```

Comment:
I could've combined all of this logic into the `useMemo`, but it's really hard to read so I opted for this setup